iphone/Maps/UI/PlacePage/PlacePageLayout/Layouts/PlacePageTrackLayout.swift

```diff
@@ -0,0 +1,127 @@
+  }()
+
+  lazy var previewViewController: PlacePagePreviewViewController = {
+    let vc = storyboard.instantiateViewController(ofType: PlacePagePreviewViewController.self)
+    vc.placePagePreviewData = placePageData.previewData
+    return vc
+  }()
+
+  lazy var placePageNavigationViewController: PlacePageHeaderViewController = {
+    return PlacePageHeaderBuilder.build(data: placePageData.previewData, delegate: interactor, headerType: .fixed)
+  }()
+
+  lazy var bookmarkViewController: PlacePageBookmarkViewController = {
+    let vc = storyboard.instantiateViewController(ofType: PlacePageBookmarkViewController.self)
+    vc.view.isHidden = true
+    vc.delegate = interactor
+    return vc
+  }()
+
+  lazy var elevationMapViewController: ElevationProfileViewController? = {
+    guard let elevationProfileData = placePageData.trackData?.elevationProfileData else {
+      return nil
+    }
+    return ElevationProfileBuilder.build(elevationProfileData: elevationProfileData, delegate: interactor)
+  }()
+
+  lazy var actionBarViewController: ActionBarViewController = {
+    let vc = storyboard.instantiateViewController(ofType: ActionBarViewController.self)
+    vc.placePageData = placePageData
+    vc.canAddStop = MWMRouter.canAddIntermediatePoint()
+    vc.isRoutePlanning = MWMNavigationDashboardManager.shared().state != .hidden
+    vc.delegate = interactor
+    return vc
+  }()
+
+  init(interactor: PlacePageInteractor, storyboard: UIStoryboard, data: PlacePageData) {
+    self.interactor = interactor
+    self.storyboard = storyboard
+    self.placePageData = data
+  }
+
+  private func configureViewControllers() -> [UIViewController] {
+    var viewControllers = [UIViewController]()
+
+    viewControllers.append(bookmarkViewController)
+    if let trackData = placePageData.trackData {
+      bookmarkViewController.bookmarkData = .track(trackData)
+      bookmarkViewController.view.isHidden = false
+    }
+
+    placePageData.onBookmarkStatusUpdate = { [weak self] in
+      guard let self = self else { return }
+      self.previewViewController.placePagePreviewData = self.placePageData.previewData
+      self.updateTrackRelatedSections()
+    }
+
+    if let elevationMapViewController {
+      viewControllers.append(elevationMapViewController)
+    }
+
+    return viewControllers
+  }
+
+  func calculateSteps(inScrollView scrollView: UIScrollView, compact: Bool) -> [PlacePageState] {
+    var steps: [PlacePageState] = []
+    let scrollHeight = scrollView.height
+    steps.append(.closed(-scrollHeight))
+    guard elevationMapViewController != nil else {
+      steps.append(.full(0))
+      return steps
+    }
+    guard let previewView = previewViewController.view else {
+      return steps
+    }
+    let previewFrame = scrollView.convert(previewView.bounds, from: previewView)
+    steps.append(.preview(previewFrame.maxY - scrollHeight))
+    if !compact {
+      steps.append(.expanded(-scrollHeight * 0.55))
```

Comment:
nit: What is this magic constant?

Reply:
The same as used in the common layout since 2020
https://github.com/organicmaps/organicmaps/blob/b03108318c85e054816588442cb5227591fa4cd6/iphone/Maps/UI/PlacePage/PlacePageLayout/Layouts/PlacePageCommonLayout.swift#L166-L175

This one helps to expand the PP to make the `ElevationProfile` partly visible.